src/connectionManager.ts

```diff
@@ -0,0 +1,264 @@
+      return Promise.reject('Unable to disconnect: already disconnecting from an instance.');
+    }
+
+    if (!this._currentConnection) {
+      return Promise.resolve(false);
+    }
+
+    // Disconnect from the active connection.
+    return new Promise<boolean>((resolve, reject) => {
+      this._disconnecting = true;
+      this._statusView.showMessage('Disconnecting from current connection...');
+
+      this._currentConnection.disconnect((err: any) => {
+        if (err) {
+          // Show an error, however we still reset the active connection to free up the extension.
+          vscode.window.showErrorMessage('An error occured while disconnecting from the current connection.');
+        } else {
+          vscode.window.showInformationMessage('MongoDB connection removed.');
+        }
+
+        this._currentConnection = null;
+        this._currentConnectionInstanceId = null;
+
+        this._disconnecting = false;
+        this._statusView.hideMessage();
+
+        return resolve(true);
+      });
+    });
+  }
+
+  public async removeMongoDBConnection(): Promise<boolean> {
+    log.info('mdb.removeMongoDBConnection command called');
+
+    // Ensure we aren't currently connecting or disconnecting.
+    if (this._connecting) {
+      return Promise.reject('Unable to remove connection: currently connecting.');
+    }
+
+    const connectionInstanceIds = Object.keys(this._connectionConfigs);
+
+    if (connectionInstanceIds.length === 0) {
+      // No active connection(s) to remove.
+      return Promise.reject('No connections to remove.');
+    }
+
+    let connectionToRemove;
+    if (connectionInstanceIds.length === 1) {
+      connectionToRemove = connectionInstanceIds[0];
+    } else {
+      // There is more than 1 possible connection to remove.
+      connectionToRemove = await vscode.window.showQuickPick(connectionInstanceIds, {
+        placeHolder: 'Choose a connection to remove...'
+      });
+    }
+
+    if (!connectionToRemove) {
+      return Promise.resolve(false);
+    }
+
+    const removeConfirmationResponse = await vscode.window.showInformationMessage(
+      `Are you sure to want to remove connection ${connectionToRemove}?`,
+      { modal: true },
+      'Yes'
+    );
+
+    if (removeConfirmationResponse !== 'Yes') {
+      return Promise.resolve(false);
+    }
+
+    if (this._currentConnection && connectionToRemove === this._currentConnectionInstanceId) {
+      await this.disconnect();
+    }
+
+    delete this._connectionConfigs[connectionToRemove];
+    vscode.window.showInformationMessage('MongoDB connection removed.');
+    return Promise.resolve(true);
+  }
+
+  // Exposed for testing.
+  public getConnections() {
+    return this._connectionConfigs;
+  }
+  public getActiveConnectionInstanceId() {
+    return this._currentConnectionInstanceId;
+  }
+  public getActiveConnection() {
+    return this._currentConnectionInstanceId;
+  }
+  public getConnnecting() {
```

Comment:
Maby `isConnecting` or something would be better? I am not sure what to expect from `getConnnecting` :)